File: unimagtienda/src/main/java/com/tienda/unimagtienda/app/product/entity/Product.java
package com.tienda.unimagtienda.app.product.entity;

import com.tienda.unimagtienda.app.category.entity.Category;
import com.tienda.unimagtienda.app.inventory.entity.Inventory;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "products")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String sku;

    @Column(nullable = false)
    private String name;

    private String description;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal price;

    @Column(nullable = false)
    private Boolean active = true;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    // Relación con Category
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id", nullable = false)
    private Category category;

    // Relación con Inventory (bidireccional)
    @OneToOne(mappedBy = "product", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private Inventory inventory;

    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}

File: fronend/ecomerce/src/features/product/type/product.ts
import type  {CategoryResponse} from "../../category/type/category.ts";


export interface ProductResponse {
    id: number;
    sku: string;
    name: string;
    description: string;
    price: string;
    active: boolean;
    createdAt: string;
    updatedAt: string;
    category: CategoryResponse;
}
export interface CreateProductRequest {
    sku: string;
    name: string;
    description?: string;
    price: number;

    categoryId: number;
}
export interface UpdateProductRequest {
    sku?: string;
    name?: string;
    description?: string;
    price?: number;
    active?: boolean;
    categoryId?: number;
}
